js/scripts.js

```diff
@@ -0,0 +1,58 @@
+var optim = document.getElementById("optimbtn")
+
+var add = document.getElementById("addbtn");
+
+var modal = document.getElementById("modal");
+
+var table = document.querySelector("table");
+
+var form = document.getElementById("form")
+
+add.onclick = function(){
+  modal.style.display = "block";
+}
+
+form.onsubmit = function(event){
+  var nameValue = document.getElementById("name").value;
+  var costValue = document.getElementById("cost").value;
+  var row = table.insertRow();
+  var cell = row.insertCell();
+  cell.innerHTML = nameValue;
+  cell = row.insertCell();
+  cell.innerHTML = costValue;
+  cell = row.insertCell();
+  cell.innerHTML = '<input type="checkbox" name="done1" value="done1">';
+  modal.style.display = "none";
+  event.preventDefault();
+}
+
+optim.onclick = function(){
```

Comment:
let's improve this. You could use an Array sort (with a custom sort function) to generate a sorted list and then iterate that list  and use the `appendChild`  function.